dnf-behave-tests/dnf/api/repo-query.feature

```diff
@@ -0,0 +1,95 @@
+@dnf5
+Feature: api: query repos
+
+
+Scenario: Construct query and filter dnf-ci-fedora repo
+Given I use repository "simple-base"
+  And I use repository "dnf-ci-fedora"
+  And I use repository "dnf-ci-fedora-updates"
+ When I execute python libdnf5 api script with setup
+      """
+      query = libdnf5.repo.RepoQuery(base)
+      query.filter_id("dnf-ci-fedora")
+      for repo in query:
+          print(repo.get_id())
+      """
+ Then the exit code is 0
+  And stdout is
+      """
+      dnf-ci-fedora
+      """
+
+
+Scenario: Construct query and filter fails due to bad argument type
+Given I use repository "simple-base"
+ When I execute python libdnf5 api script with setup
+      """
+      query = libdnf5.repo.RepoQuery(base)
+      query.filter_id(123)
+      for repo in query:
+          print(repo.get_id())
+      """
+ Then the exit code is 1
+  And stdout is empty
+  And stderr contains "TypeError: Wrong number or type of arguments for overloaded function 'RepoQuery_filter_id'."
+
+
+Scenario: Construct query and filter using QueryCmp_ICONTAINS
+Given I use repository "simple-base"
+  And I use repository "dnf-ci-fedora"
+  And I use repository "dnf-ci-fedora-updates"
+ When I execute python libdnf5 api script with setup
+      """
+      from libdnf5.common import QueryCmp_ICONTAINS
+      query = libdnf5.repo.RepoQuery(base)
+      query.filter_id("CI", QueryCmp_ICONTAINS)
+      for repo in query:
+          print(repo.get_id())
+      """
+ Then the exit code is 0
+  And stdout is
+      """
+      dnf-ci-fedora
+      dnf-ci-fedora-updates
+      """
+
+
+Scenario: Construct query and filter using QueryCmp_IGLOB
+Given I use repository "simple-base"
+  And I use repository "dnf-ci-fedora"
+  And I use repository "dnf-ci-fedora-updates"
+ When I execute python libdnf5 api script with setup
+      """
+      from libdnf5.common import QueryCmp_IGLOB
+      query = libdnf5.repo.RepoQuery(base)
+      query.filter_id(["*BASE*", "*updates*"], QueryCmp_IGLOB)
+      for repo in query:
+          print(repo.get_id())
+      """
+ Then the exit code is 0
+  And stdout is
+      """
+      simple-base
+      dnf-ci-fedora-updates
+      """
+
+
+Scenario: Construct query and filter using QueryCmp_NOT_IGLOB
+Given I use repository "simple-base"
+  And I use repository "dnf-ci-fedora"
+  And I use repository "dnf-ci-fedora-updates"
```

Comment:
Since this section is (can be) used by all of the `Scenarios` you could extract it into a `Background` at the top of the file to de-duplicate it.

Something like in https://github.com/rpm-software-management/ci-dnf-stack/blob/main/dnf-behave-tests/dnf/installonly.feature but with all three repos.

Reply:
OK, done. Thanks